src/common/CrosschainBatcher.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IGateway} from "./interfaces/IGateway.sol";
+import {ICrosschainBatcher} from "./interfaces/ICrosschainBatcher.sol";
+
+import {Auth} from "../misc/Auth.sol";
+
+/// @title  CrosschainBatcher
+/// @dev    Helper contract that enables integrations to automatically batch multiple messages. Should be used like:
+///         ```
+///         contract Integration {
+///             ICrosschainBatcher batcher;
+///
+///             function doSomething(PoolId poolId) external {
+///                 uint256 cost = batcher.batch(abi.encodeWithSelector(Integration._execute.selector, poolId));
+///             }
+///
+///             function _execute(PoolId poolId) external {
+///                 require(batcher.sender() == address(this));
+///                 // Call several hub, balance sheet, or spoke methods that trigger cross-chain transactions
+///             }
+///         }
+///         ```
+contract CrosschainBatcher is Auth, ICrosschainBatcher {
```

Comment:
@lemunozm want to add this to the deploy scripts already in this PR? Likely needed for Onno to add integration tests for the queue manager.